### DISCUSSION TITLE
Docker and portainer setup

### DISCUSSION THREAD

**Discussion by mmol67** [lardbit/nefarious#107] [Q&A] [answered]
Hello!
I'm almost a complete noob in docker/portainter. I'm trying to setup Nefarious. 
I began installing  Portainer, then set up Transmission (port 9091)  and Jackett (port 9117) containers using the instructions in linuxserver io (volumes, environment variables,...). All working ok.

I set up qbittorrent, radarr, sonarr, bazarr and all is working. 

Now I'd like to test nefarious because I think is better and easier to use.

Could you please tell me the steps to get Nefarious working using Portainer? Do I need to install a Redis container or Celery or something more? Volumes? Environment variables?

Thank you in advance

**Comment by lardbit** [comment]
Hey @friki67,

Portainer is a really nice application.  I don't use it regularly but I just tested nefarious on it and it was surprisingly straight forward.  All I had to do was paste in the `docker-compose.yml` file into the *Web Editor*.  The only caveat is that you have to combine the multiple docker compose files ( `docker-compose.yml` and `docker-compose.base.yml`) first.

To combine nefarious' compose files, run the following and copy the output to your clipboard:

    docker-compose config

Then, create a new "stack" and paste the compose output into the *Web Editor* and voi la.

You can technically skip the environment variables but you'll at least want to define `HOST_DOWNLOAD_PATH`.  See https://github.com/lardbit/nefarious/blob/master/env.template for all environment options.


![image](https://user-images.githubusercontent.com/45122868/115873201-476dbd80-a408-11eb-8f93-06b143874752.png)

**Reply by mmol67** [reply]
Thank you very much!
I've been able to install all!
I had some issues with the log options because journald driver (It seems the default in my setup) is incompatible with the max-size parameter. Removing these parameters allowed me to create the stack with all included containers.

Now I have a problem configuring Nefarious. After access the fresh created jackett container, added some public indexers, an filled the API field in Nefarious initial configuration dialog, then I press "Verify Indexers" and get a message saying "An unknown error occurred verifying jackett indexers".

So I tried changing the host field from the default "jackett" to:
1) localhost
2) the local ip for my server (192.168.99.105)
3) the name of the jackett container nefariuosstack_jackett_1
4) the ip for the docker network created (nefariousstack_default),  like 172.19.0.2

And nothing has worked.
Could you help, please?

**Reply by lardbit** [reply]
Make sure you save nefarious settings before clicking *Verify Settings* and keep the jackett host as `jackett` and likewise for `transmission`.

If you still receive that error, please paste the logs from nefarious:
  
    docker-compose logs nefarious

or you can access the logs via portainer.  That'll give me a better idea what the problem is.

**Reply by mmol67** [reply]
Save! And all ok. Thank you 👍

**Reply by lardbit** [reply]
I should just have it automatically save when verifying settings.

**Comment by lardbit** [comment, marked as answer]
This was answered
